CharlieBackend.Business/Services/StudentGroupService.cs

```diff
@@ -109,11 +109,12 @@ public async Task<Result<bool>> IsGroupNameExistAsync(string name)
             return Result<bool>.GetSuccess(res);
         }
 
-        public async Task<IList<StudentGroupDto>> GetAllStudentGroupsAsync()
+        public async Task<Result<IList<StudentGroupDto>>> GetAllStudentGroupsAsync()
```

Comment:
there's no point in using Result for list methods (since there's no validation happens)